batch_processor.go

```diff
@@ -0,0 +1,77 @@
+package failure_detector
+
+import (
+	"context"
+	"time"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// processFunc a function that processes a batch of EndpointSamples
+type processFunc func(objs []*EndpointSample)
+
+// processor retrieves EndpointSamples from the exposed channel and calls out to processFunc for processing
+type processor struct {
```

Comment:
in my own words: processor reads samples from a channel, keys them, inserts into a batch queue and runs workers processing batches